test/redirects.test.js

```diff
@@ -404,4 +405,68 @@ lab.experiment('hapi-redirect', () => {
       });
     });
   });
+
+  lab.test(' accepts a callback that adds additional dynamic routes', (done) => {
+    let count = 0;
+    server.register({
+      register: redirectModule,
+      options: {
+        log: true,
+        log404: true,
+        getRedirects(slug, redirectDone) {
+          count++;
+          return redirectDone(null, {
+            '/test': {
+              destination: `/newtest${count}`,
+              statusCode: 302
+            }
+          });
+        }
+      }
+    },
+    () => {
+      server.inject({
+        method: 'get',
+        url: '/test'
+      }, (result) => {
+        Code.expect(result.statusCode).to.equal(302);
+        Code.expect(result.headers.location).to.equal('/newtest1');
+        server.inject({
+          method: 'get',
+          url: '/test'
+        }, (result2) => {
+          Code.expect(result2.statusCode).to.equal(302);
+          Code.expect(result2.headers.location).to.equal('/newtest2');
+          done();
+        });
+      });
+    });
+  });
+
+  lab.test('emits event when redirect occurs', (done) => {

```

Comment:
can we do a quick scan of tests to make sure if there is a 404 and no redirect matches, that it just continues to send the 404

Reply:
Added a test for that here: https://github.com/firstandthird/hapi-redirects/pull/28/files#diff-86f5351edab474a541ab1bcf9f7b5a6fR484